storage/src/main/java/tech/pegasys/teku/storage/events/StorageUpdate.java

```diff
@@ -106,4 +110,8 @@ public boolean isEmpty() {
   public Map<UnsignedLong, VoteTracker> getVotes() {
     return votes;
   }
+
+  public Map<Bytes32, SlotAndBlockRoot> getStateRootsToBlockRoots() {
+    return stateRootsToBlockRoots;
+  }
```

Comment:
We should probably either name this to be clear that it only contains state roots for empty slots or update `StateTransition` to also put the state root after each processed block. I kind of like the idea of `StateTransition` putting all states and then you can avoid iterating through the hot blocks in `RocksDbDatabase` to record their state roots.  Either way is fine though - not sure if there are potential issues with doing it in `StateTransition`.

Reply:
its now called stateRoots and everything that's putting into the hot storage is putting into that list, so hopefully i've resolved this comment.